Firestore/Example/Tests/Integration/API/FIRDatabaseTests.m

```diff
@@ -931,4 +932,175 @@ - (void)testWatchStreamReconnectsAfterIdle {
   [self readSnapshotForRef:[self documentRef] requireOnline:YES];
 }
 
+
+
+- (void)testGetDocumentWhileOnlineWithDefaultGetOptions {
+  FIRDocumentReference *doc = [self documentRef];
+
+  // set document to a known value
+  NSDictionary<NSString *, id> *initialData = @{@"key" : @"value"};
+  [self writeDocumentRef:doc data:initialData];
+
+  // get doc and ensure that it exists, is *not* from the cache, and matches
+  // the initialData.
+  FIRDocumentSnapshot *result = [self readDocumentForRef:doc];
+  XCTAssertTrue(result.exists);
+  XCTAssertFalse(result.metadata.fromCache);
+  XCTAssertEqualObjects(result.data, initialData);
+}
+
+- (void)testGetDocumentWhileOfflineWithDefaultGetOptions {
+  FIRDocumentReference *doc = [self documentRef];
+
+  // set document to a known value
+  NSDictionary<NSString *, id> *initialData = @{@"key1" : @"value1"};
+  [self writeDocumentRef:doc data:initialData];
+
+  // go offline for the rest of this test
+  [self disableNetwork];
+
+  // update the doc (though don't wait for a server response. We're offline; so
+  // that ain't happening!). This allows us to further distinguished cached vs
+  // server responses below.
+  NSDictionary<NSString *, id> *newData = @{@"key2" : @"value2"};
+  [doc setData:newData completion:^(NSError *_Nullable error) {
+    XCTAssertTrue(false, "Because we're offline, this should never occur.");
+  }];
+
+  // get doc and ensure it exists, *is* from the cache, and matches the
+  // newData.
+  FIRDocumentSnapshot *result = [self readDocumentForRef:doc];
+  XCTAssertTrue(result.exists);
+  XCTAssertTrue(result.metadata.fromCache);
+  XCTAssertEqualObjects(result.data, newData);
+}
```

Comment:
I feel like these two paths are already well exercised in the rest of the test cases.

Reply:
I certainly hope so. :) I "added" these for context for this feedback review. (I'm not proposing including them in the final version, though I see that I certainly didn't make that clear.)

I've added comments to that effect, though realistically, those comments will be deleted (along with these two cases) before being uploaded to github again. (Unless I forget.)